docs/data/charts/scatter/CustomScatter.tsx

```diff
@@ -0,0 +1,72 @@
+import * as React from 'react';
+import { ScatterChart } from '@mui/x-charts/ScatterChart';
+import { useScatterSeries, useXScale, useYScale } from '@mui/x-charts/hooks';
+
+const data1 = [
+  { x: 100, y: 200, id: 1 },
+  { x: 120, y: 100, id: 2 },
+  { x: 170, y: 300, id: 3 },
+  { x: 140, y: 250, id: 4 },
+  { x: 150, y: 400, id: 5 },
+  { x: 110, y: 280, id: 6 },
+];
+const data2 = [
+  { x: 300, y: 300, id: 1 },
+  { x: 200, y: 700, id: 2 },
+  { x: 400, y: 500, id: 3 },
+  { x: 340, y: 350, id: 4 },
+  { x: 420, y: 280, id: 5 },
+];
+
+function LinkPoints({ seriesId, close }: { seriesId: string; close?: boolean }) {
+  const scatter = useScatterSeries();
+  const xScale = useXScale();
+  const yScale = useYScale();
+
+  if (!scatter) {
+    return null;
+  }
+
+  const color = scatter.series[seriesId]?.color;
+  const data = scatter.series[seriesId]?.data;
```

Comment:
nitpick, but maybe that works

```suggestion
  if (!scatter || !scatter.series[seriesId]) {
    return null;
  }

  const { color, data } = scatter.series[seriesId];
```